src/csharp/Grpc.Tools/ProtoCompilerOutputs.cs

```diff
@@ -68,8 +76,13 @@ public override bool Execute()
             // Get language-specific possible output. The generator expects certain
             // metadata be set on the proto item.
             var possible = new List<ITaskItem>();
+            var patched = new List<ITaskItem>();
             foreach (var proto in Protobuf)
             {
+                // This operates on a local copy and has no effect on the MSBuild instance!
```

Comment:
Just saw this comment.  It seems to contradict the doc comment of the method: `// Update OutputDir and GrpcOutputDir for the item and all subsequent
        // targets using this item. This should only be done if the real
        // output directories for protoc should be modified.`

If we're actually only modifying the output dir values in this instance, there should be a less confusing way of doing this (store in a different variable / metadata?)

Reply:
I might rework this comment. The `generator.PatchOutputDirectory` will patch the instance of `proto` which is local-only to avoid having to create a copied instance. `Protobuf` is only a copy based on the input and any changes done to it won't affect MSBuild.

I could store it in a different variable, but then we would either have to decide to deep-copy `proto` which decreases performance and makes no sense since it's a local-only copy already, or if we assign it to another variable but still modify `proto`, we would need to document it properly so that everyone knows that `proto` is also patched, not just the return value.

This seemed like the most obvious without side-effects.